modules/core/src/main/java/org/apache/ignite/internal/processors/platform/PlatformContextImpl.java

```diff
@@ -197,24 +197,11 @@ public PlatformContextImpl(GridKernalContext ctx, PlatformCallbackGateway gate,
         // Send node info to the native platform
         try (PlatformMemory mem0 = mem.allocate()) {
             PlatformOutputStream out = mem0.output();
-
             BinaryRawWriterEx w = writer(out);
 
-            w.writeUuid(node.id());
-            PlatformUtils.writeNodeAttributes(w, node.attributes());
-            w.writeCollection(node.addresses());
-            w.writeCollection(node.hostNames());
-            w.writeLong(node.order());
-            w.writeBoolean(node.isLocal());
-            w.writeBoolean(node.isDaemon());
-            w.writeBoolean(node.isClient());
-            w.writeObjectDetached(node.consistentId());
-            PlatformUtils.writeNodeVersion(w, node.version());
-
-            writeClusterMetrics(w, node.metrics());
+            PlatformUtils.writeNodeInfo(w, node);
```

Comment:
Let's not reuse thick client serialization logic in thin client.
This is error-prone: in thick client this does not go over the wire, but in thin it does, so we have to deal with versions and compatibility.